archinstall/lib/disk/device_handler.py

```diff
@@ -0,0 +1,593 @@
+		else:
+			log(f'Use existing device: {modification.device_path}')
+			disk = modification.device.disk
+
+		log(f'Creating partitions: {modification.device_path}')
+
+		# TODO sort by delete first
+
+		for part_mod in modification.partitions:
+			# don't touch existing partitions
+			if part_mod.exists():
+				continue
+
+			# if the entire disk got nuked then we don't have to delete
+			# any existing partitions anymore because they're all gone already
+			requires_delete = modification.wipe is False
+			self._perform_partitioning(part_mod, modification.device, disk, requires_delete=requires_delete)
+
+		self.partprobe(modification.device.device_info.path)
+
+	def mount(
+		self,
+		dev_path: Path,
+		target_mountpoint: Path,
+		mount_fs: Optional[str] = None,
+		create_target_mountpoint: bool = True,
+		options: List[str] = []
+	):
+		if create_target_mountpoint and not target_mountpoint.exists():
+			target_mountpoint.mkdir(parents=True, exist_ok=True)
+
+		if not target_mountpoint.exists():
+			raise ValueError('Target mountpoint does not exist')
+
+		lsblk_info = get_lsblk_info(dev_path)
+		if target_mountpoint in lsblk_info.mountpoints:
+			log(f'Device already mounted at {target_mountpoint}')
+			return
+
+		str_options = ','.join(options)
+		str_options = f'-o {str_options}' if str_options else ''
+
+		mount_fs = f'-t {mount_fs}' if mount_fs else ''
+
+		command = f'mount {mount_fs} {str_options} {dev_path} {target_mountpoint}'
+
+		log(f'Mounting {dev_path}: command', level=logging.DEBUG)
+
+		try:
+			result = SysCommand(command)
+			if result.exit_code != 0:
+				raise DiskError(f'Could not mount {dev_path}: {command}\n{result.decode()}')
+		except SysCallError as err:
+			raise DiskError(f'Could not mount {dev_path}: {command}\n{err.message}')
+
+	def umount(self, mountpoint: Path, recursive: bool = False):
+		try:
+			lsblk_info = get_lsblk_info(mountpoint)
+		except SysCallError as ex:
+			# this could happen if before partitioning the device contained 3 partitions
+			# and after partitioning only 2 partitions were created, then the modifications object
+			# will have a reference to /dev/sX3 which is being tried to umount here now
+			if 'not a block device' in ex.message:
+				return
+
+			raise ex
+
+		if len(lsblk_info.mountpoints) > 0:
+			log(f'Partition {mountpoint} is currently mounted at: {[str(m) for m in lsblk_info.mountpoints]}', level=logging.DEBUG)
+
+			for mountpoint in lsblk_info.mountpoints:
+				log(f'Unmounting mountpoint: {mountpoint}', level=logging.DEBUG)
+
+				command = 'umount'
+
+				if recursive:
+					command += ' -R'
+
+				try:
+					SysCommand(f'{command} {mountpoint}')
+				except SysCallError as err:
+					# Without to much research, it seams that low error codes are errors.
+					# And above 8k is indicators such as "/dev/x not mounted.".
+					# So anything in between 0 and 8k are errors (?).
+					if err and err.exit_code and 0 < err.exit_code < 8000:
```

Comment:
See f611310

Reply:
@codefiles thanks, I updated it to align with the master version